src/transaction/transaction_master_group_complete_manager.cpp

```diff
@@ -96,7 +96,8 @@ namespace cubtx
     /* This function is called after adding a transaction to the current group.
      * Currently, we wakeup GC thread when first transaction is added into current group.
      */
-    if (get_current_group ().get_container ().size () == 1)
+    if (is_latest_closed_group_completed ()
+        && get_current_group ().get_container ().size () == 1)
```

Comment:
Safer to consider at least one:
```suggestion
        && get_current_group ().get_container ().size () >= 1)
```

Reply:
As I told to Razvan, I currently know is that the common functionality is to have exactly m_current_group_min_transactions. However, special cases (on master probably) may be greater than m_current_group_min_transactions. 
For simplicity, up to knowing exactly what expects every node, I added all cases in that functions.

For now I will add TODO, to be clear what slave expects.

About get_current_group ().get_container ().size () >= 1. I considered this case. But, I think that we don't need currently and we don't want to keep mutex longer time, if we don't need. Let's not forget that mainly the GC thread does wakeup at notify.